Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -0,0 +1,384 @@
+        puts "No newer upstream versions."
+      end
+
+      puts JSON.generate(formulae_checked.compact) if args.json?
+    end
+
+    def formula_name(formula, args:)
+      args.full_name? ? formula.full_name : formula.name
+    end
+
+    def status_hash(formula, status_str, messages = nil, args:)
+      status_hash = {
+        "formula" => formula_name(formula, args: args),
+        "status"  => status_str,
+      }
+      status_hash["messages"] = messages.presence
+
+      if args.verbose?
+        status_hash["meta"] = {
+          "livecheckable" => formula.livecheckable?,
+        }
+        status_hash["meta"]["head_only"] = !formula.stable? unless formula.stable?
+      end
+
+      status_hash
+    end
+
+    def skip_conditions(formula, args:)
+      if formula.deprecated? && !formula.livecheckable?
+        return status_hash(formula, "deprecated", args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : deprecated"
+          return
+        end
+      end
+
+      if formula.to_s.include?("@") && !formula.livecheckable?
+        return status_hash(formula, "versioned", args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : versioned"
+          return
+        end
+      end
+
+      if !formula.stable? && !formula.any_version_installed?
+        head_only_msg = "HEAD only formula must be installed to be livecheckable"
+        return status_hash(formula, "error", [head_only_msg], args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : #{head_only_msg}"
+          return
+        end
+      end
+
+      is_gist = formula.stable&.url&.include?("gist.github.com")
+      if formula.livecheck.skip? || is_gist
+        skip_msg = if formula.livecheck.skip_msg.is_a?(String) &&
+                      !formula.livecheck.skip_msg.blank?
+          formula.livecheck.skip_msg.to_s
+        elsif is_gist
+          "Stable URL is a GitHub Gist"
+        else
+          ""
+        end
+
+        return status_hash(formula, "skipped", (skip_msg.blank? ? nil : [skip_msg]), args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : skipped" \
+              "#{" - #{skip_msg}" unless skip_msg.blank?}"
+          return
+        end
+      end
+
+      false
+    end
+
+    def print_latest_version(info, args:)
+      formula_s = "#{Tty.blue}#{info["formula"]}#{Tty.reset}"
+      formula_s += " (guessed)" if !info["meta"]["livecheckable"] && args.verbose?
+
+      current_s =
+        if info["version"]["newer_than_upstream"]
+          "#{Tty.red}#{info["version"]["current"]}#{Tty.reset}"
+        else
+          info["version"]["current"]
+        end
```

Comment:
```suggestion
      current_s = if info["version"]["newer_than_upstream"]
        "#{Tty.red}#{info["version"]["current"]}#{Tty.reset}"
      else
        info["version"]["current"]
      end
```

Would be nice to use this format everywhere.